components/script/dom/readablestream.rs

```diff
@@ -423,6 +431,47 @@ impl ReadableStream {
 
         // TODO: react to sourceCancelPromise.
     }
+
+    /// <https://streams.spec.whatwg.org/#readable-stream-reader-generic-release>
+    pub fn release_lock(
+        &self,
+        closed_promise: &Promise,
+        read_requests: DomRefCell<VecDeque<ReadRequest>>,
+    ) {
+        // step 2.1.3
+        assert!(self.is_locked());
+
+        if self.is_readable() {
+            // step 2.1.4
+            closed_promise.reject_error(Error::Type("stream state is not readable".to_owned()));
+        } else {
+            // step 2.1.5
+            closed_promise.reject_error(Error::Type(
+                "Cannot release lock due to stream state.".to_owned(),
+            ));
+        }
+
+        // step 2.1.6
+        closed_promise.set_promise_is_handled();
+
+        // step 2.1.7
+        self.perform_release_steps();
+
+        // step 2.1.8 & 2.1.9
+        self.set(None);
+
+        // step  3.1
+        let error = Error::Type("No chunks are available because the stream is errored".to_owned());
+
+        let read_requests_cloned = read_requests.clone();
+        // <https://streams.spec.whatwg.org/#abstract-opdef-readablestreamdefaultreadererrorreadrequests>
+        // step 3.2
+        read_requests.borrow_mut().clear();
+        // step 3.3
+        for request in read_requests_cloned.borrow_mut().drain(0..) {
```

Comment:
I don't think you need the two steps above, and here you should do something like [here](https://github.com/servo/servo/blob/1ff141d486ddd0f28c1af45c9f873154138c9b47/components/script/dom/readablestreamdefaultreader.rs#L147), because the error steps can call back into the stream and cause a borrow error(but see above comment, you should re-use the existing code anyway).